### PR TITLE
CHIPS AND DIP and also burritos

### PR DESCRIPTION
**Pull request by BurgerLUA**
# CHIPS AND DIP
CHIPS

Adds tortillas, tortilla chips and nacho tips, along with 3 different types of tips: Guac, Salsa, and Cheese dip.

Tortillas can be made by microwaving 5 units flour, 5 units water, and 1 unit salt in the microwave. From there, they can be microwaved again with an additional unit of salt to create a basket of plain chips. Combining plain chips and cheese in the microwave can make a plate of nachos.

Dips can also be made in the microwave. All dips require ingredients from the garden.

You can grab a single chip or nacho by activating a basket of plain chips or a plate of nachos. From there, you can dip that single chip into the dip. Each plate of chips contains about 20 chips, and each dip contains about 20 uses. You can double dip, if you're a spawn of satan.

# also burritos

also burritos

Adds meat burritos, vegan burritos, spicy meat burritos, meat cheese burritos, spicy meat cheese burritos, mystery meat burritos, and el diablo burritos.

## TODO:
- Possibly work on custom dip options.
- Gather feedback on code and recipes values.

### Additional changes.
- Added my snowflake drink, Dr. Daniels, which consists of Diet Dr Gibb (Dr. Gibb, Water), Honey, and whiskey. Perhaps this can set a precedent for players asking for custom food/drink requests if they provide the sprites like there are for character items.
- Added an alternative enchiladas and taco recipe, which takes 1 tortilla instead of corn/dough slice.

### CODE REVIEW

File: code/modules/food/recipes_microwave.dm [hunk 1]
```diff
@@ -903,3 +900,65 @@ I said no!
...
+	items = list(
+		/obj/item/weapon/reagent_containers/food/snacks/tofu
+	)
+	result = /obj/item/weapon/reagent_containers/food/snacks/dip/guac
+	reagent_mix = RECIPE_REAGENT_SUM
+
+/datum/recipe/cheesesauce
+	fruit = list("chili" = 1, "tomato" = 1)
+	reagents = list("spacespice" = 1, "blackpepper" = 1,"sodiumchloride" = 1)
+	items = (
```
Comment (skull132) [inline code/modules/food/recipes_microwave.dm]: How the fuck does this even compile. 🤔 
This should be:
```DM
	items = list(
		/obj/item/weapon/reagent_containers/food/snacks/cheesewedge
	)
```

Comment (BurgerLUA) [reply code/modules/food/recipes_microwave.dm]: oh that explains why it wasn't passing travis ty papa

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -4818,4 +4819,133 @@
...
+	desc = "The mystery is, why aren't you BSAing it?"
+	icon_state = "burrito_mystery"
+	bitesize = 5
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("regret" = 1)
+	nutriment_amt = 1
+/obj/item/weapon/reagent_containers/food/snacks/burrito_mystery/Initialize()
+	reagents.add_reagent("protein", 4)
+	. = ..()
+	var/mysteryselect = pick(1,2,3,4,5,6,7,8,9,10) // what the fuck is this
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: or just `rand(1,10)`.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -4818,4 +4819,133 @@
...
+	switch(mysteryselect)
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Above statement could be inlined into this switch if it's not used anywhere else.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 4]
```diff
@@ -4818,4 +4819,133 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/burrito_mystery
+	name = "mystery meat burrito"
+	desc = "The mystery is, why aren't you BSAing it?"
+	icon_state = "burrito_mystery"
+	bitesize = 5
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("regret" = 1)
+	nutriment_amt = 1
+/obj/item/weapon/reagent_containers/food/snacks/burrito_mystery/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Should be a newline between separate definitions.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 5]
```diff
@@ -4818,4 +4819,133 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/burrito_hell
+	name = "el diablo"
+	desc = "Meat, melted cheese, and a fuckload of chilis packed in a flour tortilla. The chaplain's office is west of the kitchen."
+	icon_state = "burrito_hell"
+	bitesize = 4
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("tortilla" = 1)
+	nutriment_amt = 6
+/obj/item/weapon/reagent_containers/food/snacks/burrito_hell/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Should be a newline between separate definitions.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 6]
```diff
@@ -4818,4 +4819,133 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/burrito_cheese_spicy
+	name = "spicy cheese meat burrito"
+	desc = "Meat, melted cheese, and chilis wrapped in a flour tortilla. Medical is north of the washrooms."
+	icon_state = "burrito_cheese_spicy"
+	bitesize = 4
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("tortilla" = 1)
+	nutriment_amt = 6
+/obj/item/weapon/reagent_containers/food/snacks/burrito_cheese_spicy/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Should be a newline between separate definitions.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 7]
```diff
@@ -4818,4 +4819,133 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/burrito_cheese
+	name = "meat cheese burrito"
+	desc = "Meat and melted cheese wrapped in a flour tortilla. Do not feed to Skrell."
+	icon_state = "burrito_cheese"
+	bitesize = 4
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("tortilla" = 1)
+	nutriment_amt = 6
+/obj/item/weapon/reagent_containers/food/snacks/burrito_cheese/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Should be a newline between separate definitions.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 8]
```diff
@@ -4818,4 +4819,133 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/burrito_spicy
+	name = "spicy meat burrito"
+	desc = "Meat and chilis wrapped in a flour tortilla. Washrooms are north of the kitchen."
+	icon_state = "burrito_spicy"
+	bitesize = 4
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("tortilla" = 1)
+	nutriment_amt = 6
+/obj/item/weapon/reagent_containers/food/snacks/burrito_spicy/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Should be a newline between separate definitions.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 9]
```diff
@@ -4818,4 +4819,133 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/burrito_vegan
+	name = "vegan burrito"
+	desc = "Tofu, carrots, and cabbage wrapped in a flour tortilla. Those seen with this food object are Valid."
+	icon_state = "burrito_vegan"
+	bitesize = 4
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("tortilla" = 1, "lettuce" = 1, "carrot" = 1)
+	nutriment_amt = 12
+/obj/item/weapon/reagent_containers/food/snacks/burrito_vegan/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Should be a newline between separate definitions.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 10]
```diff
@@ -4818,4 +4819,133 @@
...
+//burritos
+/obj/item/weapon/reagent_containers/food/snacks/burrito
+	name = "meat burrito"
+	desc = "Meat wrapped in a flour tortilla. It's a burrito by definition."
+	icon_state = "burrito"
+	bitesize = 4
+	center_of_mass = list("x"=16, "y"=16)
+	nutriment_desc = list("tortilla" = 1)
+	nutriment_amt = 6
+/obj/item/weapon/reagent_containers/food/snacks/burrito/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Should be a newline between separate definitions.

**Comment by BurgerLUA** [comment]
Removed WIP tag. Can't really think of anything else I could add without delving into custom food.

**Comment by ProbablyMenown** [comment]
This is amazing. I want this badly.

**Pull request merged**